mempool/src/tests/test_framework.rs

```diff
@@ -0,0 +1,516 @@
+        mempool_notifications::new_mempool_notifier_listener_pair();
+
+    let mempool = Arc::new(Mutex::new(CoreMempool::new(&config)));
+    let vm_validator = Arc::new(RwLock::new(MockVMValidator));
+    let db_rw = Arc::new(RwLock::new(DbReaderWriter::new(MockDbReaderWriter)));
+    let db_ro = Arc::new(MockDbReaderWriter);
+
+    let mut event_subscriber = EventSubscriptionService::new(ON_CHAIN_CONFIG_REGISTRY, db_rw);
+    let reconfig_event_subscriber = event_subscriber.subscribe_to_reconfigurations().unwrap();
+
+    start_shared_mempool(
+        runtime.handle(),
+        &config,
+        mempool.clone(),
+        network_handles,
+        ac_endpoint_receiver,
+        consensus_events,
+        mempool_listener,
+        reconfig_event_subscriber,
+        db_ro,
+        vm_validator,
+        vec![sender],
+        peer_metadata_storage,
+    );
+
+    (
+        ac_endpoint_sender,
+        consensus_sender,
+        mempool_notifier,
+        mempool,
+    )
+}
+
+fn node_runtime(node_id: NodeId) -> Arc<Runtime> {
+    Arc::new(
+        tokio::runtime::Builder::new_multi_thread()
+            .thread_name(format!("node-{:?}", node_id))
+            .enable_all()
+            .build()
+            .expect("[test-framework] failed to create runtime"),
+    )
+}
+
+fn mpsc_channel<T>() -> (
+    futures::channel::mpsc::Sender<T>,
+    futures::channel::mpsc::Receiver<T>,
+) {
+    futures::channel::mpsc::channel(1_024)
+}
+
+pub fn test_transaction(seq_num: u64) -> TestTransaction {
+    TestTransaction::new(1, seq_num, 1)
+}
+
+pub fn test_transactions(start: u64, num: u64) -> Vec<TestTransaction> {
+    let mut txns = vec![];
+    for seq_num in start..start.checked_add(num).unwrap() {
+        txns.push(test_transaction(seq_num))
+    }
+    txns
+}
+
+pub fn block_contains_only_test_transactions(
+    block: &[SignedTransaction],
+    txns: &[TestTransaction],
+) -> bool {
+    txns.iter()
+        .all(|txn| block_contains_test_transaction(block, txn))
+        && block.len() == txns.len()
+}
+
+pub fn block_contains_any_test_transactions(
+    block: &[SignedTransaction],
+    txns: &[TestTransaction],
+) -> bool {
+    txns.iter()
+        .any(|txn| block_contains_test_transaction(block, txn))
+        && block.len() == txns.len()
+}
+
+fn block_contains_test_transaction(block: &[SignedTransaction], txn: &TestTransaction) -> bool {
+    block.iter().any(|signed_txn| {
+        signed_txn.sequence_number() == txn.sequence_number
+            && signed_txn.sender() == TestTransaction::get_address(txn.address)
+    })
+}
+
+pub async fn send_message(
+    node: &mut MempoolNode,
+    protocol_id: ProtocolId,
+    network_id: NetworkId,
+    remote_peer_id: PeerId,
+    txns: &[TestTransaction],
+) {
+    let inbound_handle = node.node_inbound_handle(network_id);
+    let request_id = Vec::new();
+    let msg = MempoolSyncMsg::BroadcastTransactionsRequest {
+        request_id: request_id.clone(),
```

Comment:
Why clone and not just `Vec::new()`?

Reply:
This needs to be generated differently between requests.  let me fix that